tests/e2e/vmlistener_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite holds testcases related with vm listener functionality.
+// In case of vm/esx host restart vDVS cleans up stale attachment details
+// the volume.
+
+package e2e
+
+import (
+	"log"
+	"os"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/esxcli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/govc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+)
+
+type VMListenerTestParams struct {
+	volumeName     string
+	dockerHostIP   string
+	dockerHostName string
+	containerName  string
+	esxIP          string
+}
+
+func (s *VMListenerTestParams) SetUpSuite(c *C) {
+	s.dockerHostIP = os.Getenv("VM2")
+	s.dockerHostName = govc.RetrieveVMNameFromIP(s.dockerHostIP)
+	s.esxIP = os.Getenv("ESX")
+}
+
+func (s *VMListenerTestParams) SetUpTest(c *C) {
+	s.volumeName = inputparams.GetVolumeNameWithTimeStamp("vmListenerTest")
```

Comment:
Nit: let's keep the naming consistent with other test groups:
vmListenerTest => vmlistener_test